design/helm.md

```diff
@@ -0,0 +1,145 @@
+# Helm Addons
+
+To be an addon in a Kurl cluster, they must be contributed into the repo and maintained across new breaking changes and new versions.
+This proposal will give users the flexibility to specify their own addons through a helm chart support in the Kurl installer spec.
+
+
+## Goals
+- Allow any Helm chart to be specified as part of a Kurl spec.
+
+## Non Goals
+
+- Verify authenticity, trustworthyness and "goodness" of helm addons.
+- Validate Helm values.
+
+## Background
+
+Users of Kurl have a limited set of available cluster addons and often want to include ones that are unsupported.
+There is friction in the [kurl development process](https://github.com/replicatedhq/kURL/blob/master/test/dev.md), [contributing new addons](https://kurl.sh/docs/add-on-author/) and addon maintanence. 
+
+Helm is a popular tool for managing deployment configuration that allows application/addon developers or the community to maintain a consistent interface for deployment through a chart.
+By allowing Kurl users to specifify a chart and its values, we can allow customization of kutl with new addons while delegating addons support and maintance to its community.
+
+## High-Level Design
+
+Kurl will bundle the Helm binary into the common tools bundle already delivering krew and kubectl plugins. The install.sh script will perform the following in the online case:
+1. Unarchive Helm and move into the `$PATH` and `/usr/bin`
+1. For each release, check if the release exists. 
+1. If the release exists, perform a `helm upgrade`.
+1. If the release doesn't exist, perform a `helm install`
+
+For airgapped the chart and artifacts need to be packaged into a bundle. The builder should perform the following:
+1. `helm pull` the repo
+1. `helm template` to render the chart with the provided values
+1. Generate an adhoc `manifest` file by parsing `image` specs from the rendered yaml.
+1. Generate a bundle including the chart + containers
+
+In the airgapped case, install.sh will perform the following:
+1. Unpack the helm bundle on initialization
+1. Load docker containers as part of the initialization
+1. Unarchive the chart and install with the installer spec values during the install phase
+
+## Detailed Design
+
+The yaml design is informed by the [Flux Helm Operator CRD](https://github.com/fluxcd/helm-operator/blob/master/docs/references/helmrelease-custom-resource.md). 
+Another spec is [Rancher's Helm CRD](/var/lib/rancher/rke2/server/manifests/) that works with k3s and RKE2.
+
+The proposal is to implement option1 in the updated Kurl Installer Spec:
+```yaml
+apiVersion: "cluster.kurl.sh/v1beta1"
+kind: "Installer"
+metadata: 
+  name: "latest"
+spec: 
+  kubernetes: 
+    version: "latest"
+  weave: 
+    version: "latest"
+  containerd: 
+    version: "latest"
+  helm:                 # Optional Future Capability
```

Comment:
Random thought, but we could have a `helmfile` plugin (https://github.com/roboll/helmfile) and just provide configuration to it as a single "addon". Not saying that's better in any sense, but it's pushing the abstraction down one layer and leveraging even more open source.

Reply:
That seems like a viable option. I'm mostly neutral in opinion. I guess consumers will have to learn our syntax or the helmfile syntax which is the same burden. I kind of like having a simpler facade to helm that we manage and having one less moving part to deliver.  If you have strong feelings I think it's not that much of a change.

I'll at least identify it in the alternatives section for now.